packages/rpc/src/batchQuery.ts

```diff
@@ -65,13 +65,19 @@ export const batchVstorageQuery = (
           }
 
           const value = JSON.parse(data.value);
+          const latestValueStr = Object.hasOwn(value, 'values')
+            ? value.values[value.values.length - 1]
+            : value;
+          const parsed = JSON.parse(latestValueStr);
+          const unserialized = Object.hasOwn(parsed, 'slots')
+            ? unserialize(parsed)
```

Comment:
Please correct the function name to "unmarshal". It's a confusion that we fixed in Endo. Now is a good time to fix it here because this code is carefully separating the parsing step.

Reply:
Done